### PR TITLE
MINT-2010 add LHT

### PR DESCRIPTION
**Pull request by alexander-elgin**

### CODE REVIEW

File: packages/core/redux/ethereumLikeBlockchain/thunks.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    return dispatch(processTransaction({
+      web3: getWeb3(state),
+      entry: pendingEntrySelector(entry.tx.from, entry.key)(state),
+      signer,
+    }))
+  }
+)
+
+export const estimateEthereumLikeBlockchainGas = (tx, feeMultiplier, getWeb3, duckId, getEstimateGasRequestFieldSet) => (
```
[the remark this answers is not in the document]
Comment (e-maximus) [reply packages/core/redux/ethereumLikeBlockchain/thunks.js]: I completely agreed with Ivan. There are many unnecessary things that could be done different way. And I would think again about EthereumLike folder

File: packages/core/services/signers/EthereumLikeBlockchainMemoryDevice.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import bip39 from 'bip39'
+import EventEmitter from 'events'
+import hdKey from 'ethereumjs-wallet/hdkey'
+import Accounts from 'web3-eth-accounts'
+import { WALLET_HD_PATH } from '@chronobank/login/network/constants'
+import { WALLET_TYPE_MEMORY } from '../../models/constants/AccountEntryModel'
+
+export const DEFAULT_PATH = `m/44'/60'/0'/0/0`
+
+export default class EthereumMemoryDevice extends EventEmitter {
```
[the remark this answers is not in the document]
Comment (alexander-elgin) [reply packages/core/services/signers/EthereumLikeBlockchainMemoryDevice.js]: It is required to make the Ethereum and Labor Hour independent. They refer to the same implementation but they are independent from each other.

File: packages/core/services/signers/EthereumMemoryDevice.js [hunk 3]
```diff
@@ -3,112 +3,7 @@
...
-    const hdWallet = hdKey.fromMasterSeed(mnemonic)
-    const w = hdWallet.derivePath(walletPath).getWallet()
-    const account = accounts.privateKeyToAccount(`0x${w.getPrivateKey().toString('hex')}`)
-    wallet.add(account)
-
-    return wallet[0]
-  }
-}
+export { DEFAULT_PATH } from './EthereumLikeBlockchainMemoryDevice'
+export default class EthereumMemoryDevice extends EthereumLikeBlockchainMemoryDevice {}
```
[the remark this answers is not in the document]
Comment (alexander-elgin) [reply packages/core/services/signers/EthereumMemoryDevice.js]: Ethereum & Labor Hour blockchain should be independent from each other. They have absolutely identical functionality thus they should use the same implementation.

File: packages/login/network/LaborHourNode.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2017–2018, LaborX PTY
+ * Licensed under the AGPL Version 3 license.
+ */
+
+import axios from 'axios'
+import { NETWORK_MAIN_ID } from './settings'
+import EthereumMiddlewareNode from './EthereumMiddlewareNode'
+
+// TODO @dkchv: update to actual config
```
[the remark this answers is not in the document]
Comment (alexander-elgin) [reply packages/login/network/LaborHourNode.js]: The labor hour Node was copied from the Ethereum Node. The original TODO as not addressed to me. I don’t know what it is about. The change required for Ethereum could be also required for labor hour. Hence I kept the TODO

**Comment by wAISw** [comment]
Data from this pr was merged in #862

**Pull request closed without merging**